src/main/java/com/iexec/common/chain/eip712/EIP712Domain.java

```diff
@@ -64,21 +62,22 @@ public EIP712Domain(String name, String version, long chainId, String verifyingC
         this.version = version;
         this.chainId = chainId;
         this.verifyingContract = verifyingContract;
-
-        this.initTypes();
     }
 
-    private void initTypes() {
-        this.types.addAll(Arrays.asList(
+    @JsonIgnore
+    public List<TypeParam> getTypes() {
```

Comment:
Not sure whether `types` are used, but this change would lead to recomputing `types` value each time it is needed. Isn't it an issue? It is used a single time?